src/main/java/pl/wedding/management/service/user/InMemoryRepository.java

```diff
@@ -0,0 +1,76 @@
+package pl.wedding.management.service.user;
+
+import pl.wedding.management.service.user.model.User;
+
+import java.util.*;
+import java.util.function.Function;
+
+final class InMemoryRepository implements UserRepository {
+
+    private static final Set<User> MEMORY_USER_REPOSITORY = new HashSet<>();
+
+    @Override
+    public Optional<User> findUserById(Integer id) {
+
+        return findByObjectValue(id, User::getUserId);
+    }
+
+    @Override
+    public Optional<User> findUserByEmail(String email) {
+
+        return findByObjectValue(email, User::getEmail);
+    }
+
+    @Override
+    public Integer addUser(User user) {
+
+        if(MEMORY_USER_REPOSITORY.contains(user)){
+            throw new IllegalStateException("User already exists!");
+        }
+
+        User userWithId = User.builder()
+                .userId(getLastIndexOfRepository())
```

Comment:
It's not last index. It's *next* 'index'. By this method name, I could assume, that: 
```
MEMORY_USER_REPOSITORY.toArray(new String[]{})[getLastIndexOfRepository()];
``` 
should be valid expression. Try it.
Notice that (array)lists and arrays are indexed from 0, and having one element in them, index is 0, and size is 1.

Reply:
I just didn't name this method in right way.